src/widgets/radio.jsx

```diff
@@ -574,17 +643,32 @@ var RadioEditor = React.createClass({
 
     onDelete: function(choiceIndex, e) {
         e.preventDefault();
+
         var choices = this.props.choices.slice();
+        var deleted = choices[choiceIndex];
+
         choices.splice(choiceIndex, 1);
-        this.props.onChange({choices: choices});
+
+        this.props.onChange({
+            choices: choices,
+            hasNoneOfTheAbove: !( deleted.isNoneOfTheAbove || !this.props.hasNoneOfTheAbove )
+        });
     },
 
-    addChoice: function(e) {
+    addChoice: function(noneOfTheAbove, e) {
         e.preventDefault();
 
         var choices = this.props.choices;
-        this.props.onChange({choices: choices.concat([{}])}, () => {
-            this.refs["editor" + choices.length].focus();
+        var newChoice = { isNoneOfTheAbove: !!noneOfTheAbove }
+        var insertIndex = choices.length - ( this.props.hasNoneOfTheAbove | 0 );
```

Comment:
No it was actually intentional to use the bitwise operator as a way to cast the boolean to an integer (it's the [fastest way](http://jsperf.com/boolean-int-conversion/2)), but I changed it anyway to the logical operator as it is more readable